torch/_dynamo/variables/inline_helper.py

```diff
@@ -0,0 +1,159 @@
+                not in tracer_to_used_names[tx.output.current_tracer].keys()
+            ):
+                tracer_to_used_names[tx.output.current_tracer][base_module_key] = 0
+
+            count = tracer_to_used_names[tx.output.current_tracer][base_module_key]
+            tracer_to_used_names[tx.output.current_tracer][base_module_key] += 1
+            unique_module_key = (
+                base_module_key if count == 0 else f"{base_module_key}_{count}"
+            )
+            node.meta["source_fn_stack"][-1] = (
+                unique_module_key,
+                type(module_vt.module),
+            )
+        # remove the additional stack trace caused by fwd inlining
+        if "stack_trace" in node.meta and len(node.meta["stack_trace"]) > 0:
+            splited = node.meta["stack_trace"].split("\n")
+            # handle the cases where make_fx is called.
+            if len(splited) > 7 and "_dynamo/variables/inline_helper.py" in splited[-5]:
+                node.meta["stack_trace"] = "\n".join(splited[:-7]) + "\n"
+            # handle the case for lifted parameters.
+            elif (
+                len(splited) > 4
+                and "return forward_call(*args, **kwargs)" in splited[-2]
+            ):
+                node.meta["stack_trace"] = "\n".join(splited[:-3]) + "\n"
+
+
+code_to_Fx = {}
+
+
+def decompose_and_inline_function_with_makefx(tx, fn, args, kwargs):
+    from functorch import make_fx
+
+    from torch._dispatch.python import enable_python_dispatcher
+    from torch._guards import detect_fake_mode
+    from .base import MutableLocal
+    from .builder import SourcelessBuilder
+    from .dicts import ConstDictVariable
+    from .lists import BaseListVariable
+
+    # convert arguments from VariableTracker to fake tensors + constants again
+    fake_value_args = []
+    for arg in args:
+        fake_value_args.append(vt_to_fake_helper(arg, tx))
+
+    fake_value_kwargs = {}
+    for key, value in kwargs.items():
+        fake_value_kwargs[key] = vt_to_fake_helper(value, tx)
+
+    # Wrap the function before calling make_fx to avoid make_fx modify the kwargs's key.
+    def wrapper_fn(fn):
+        def inner(arg, kwargs):
+            return fn(*arg, **kwargs)
+
+        return inner
+
+    wrapped_fn = wrapper_fn(fn)
+    fake_mode = detect_fake_mode(fake_value_args)
+
+    with fake_mode:
+        with enable_python_dispatcher():
+            fx_g = make_fx(wrapped_fn, pre_dispatch=True)(
+                fake_value_args, fake_value_kwargs
+            )
+
+    # this is a hack, we want to access `.code` here to trigger the `real_recompile`
+    # in case this is `_lazy_graph_module`. This will aovid us trying to inline the
+    # `_LazyGraphModule._lazy_forward`(in the skip list) below.
+    code = fx_g.code
+
+    # make_fx on the same nn_module we will create function with different names.
+    # SpeculationLog will replay the dynamo tracing upon graph break and it expects
```

Comment:
@mlazos for any thoughts on making this work with speculation log. I think another way to do it is to give control on the allocated name from make_fx, so we can force a stable name.